app/controllers/concerns/discardable_controller.rb

```diff
@@ -0,0 +1,16 @@
+module DiscardableController
+  # Override default Active Admin destroy method using Discardable Model.
+  # It sets discarded_at for entity and for its dependent resources.
+  # Redirect to resource list with proper message
+  def destroy
+    entity_name = ActiveSupport::Inflector.singularize(controller_name).camelize
+    entity_destroy_class = "::Command::Destroy::#{entity_name}".constantize
```

Comment:
can we catch possible `NameError`s here and re-throw ex with more context?

Reply:
@kowal done